fe/fe-core/src/main/java/org/apache/doris/nereids/rules/mv/SelectRollupWithAggregate.java

```diff
@@ -172,23 +183,33 @@ public List<Rule> buildRules() {
     /**
      * Select candidate rollup ids.
      * <p>
-     * todo: 0. turn off pre agg, checking input aggregate functions and group by expressions, etc.
+     * 0. turn off pre agg, checking input aggregate functions and group by expressions, etc.
      * 1. rollup contains all the required output slots.
      * 2. match the most prefix index if pushdown predicates present.
+     * 3. sort the result matching rollup index ids.
      */
-    private List<Long> selectCandidateRollupIds(
-            LogicalOlapScan olapScan,
+    private Pair<PreAggStatus, List<Long>> selectCandidateRollupIds(
+            LogicalOlapScan scan,
             Set<Slot> requiredScanOutput,
             List<Expression> predicates,
             // not used now, reserved for checking aggregate function type match.
-            List<AggregateFunction> aggregateFunctions) {
-        Preconditions.checkArgument(Sets.newHashSet(olapScan.getOutput()).containsAll(requiredScanOutput),
-                "Scan's output should contains all the input required scan output.");
+            List<AggregateFunction> aggregateFunctions,
+            List<Expression> groupingExprs) {
+        Preconditions.checkArgument(Sets.newHashSet(scan.getOutput()).containsAll(requiredScanOutput),
```

Comment:
we have a new interface to get output set, however, it returns a set of exprid. I don't know if you can use it directly here

Reply:
fixed